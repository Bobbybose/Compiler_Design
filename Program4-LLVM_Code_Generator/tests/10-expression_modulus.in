int a = 2;
char c = 'g';
int d = c % a;
return d;
// Should return 1