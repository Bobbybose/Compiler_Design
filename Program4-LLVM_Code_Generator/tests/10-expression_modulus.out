target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

define dso_local i32 @main() {
  %t1 = alloca i32
  store i32 2, i32* %t1
  %t2 = alloca i8
  store i8 103, i8* %t2
  %t3 = load i32, i32 *%t1
  %t4 = load i8, i8 *%t2
  %t5 = sext i8 %t4 to i32
  %t6 = srem i32 %t3, %t5
  %t7 = alloca i32
  store i32 %t6, i32* %t7
  ret i32 0
}
